core/shared/platform/include/platform_api_extension.h

```diff
@@ -195,6 +195,93 @@ os_cond_signal(korp_cond *cond);
 int
 os_cond_broadcast(korp_cond *cond);
 
+/**
+ * Creates a new POSIX semaphore or opens an existing
```

Comment:
Had better use `POSIX-like semaphore`, because in some RTOS it may not be a real POSIX semaphore